packages/lib/src/services/app-hosting/__tests__/provisioner-core.test.ts

```
@@ -0,0 +1,211 @@
+describe('flyAppNameFor', () => {
+  assert({
+    given: 'a published-app row id',
+    should: 'derive the Fly app name deterministically from it',
+    actual: flyAppNameFor('abc123'),
+    expected: 'pgs-app-abc123',
+  });
+
+  assert({
+    given: 'the same id twice',
+    should: 'produce the same name, so a retried provision targets the same Fly app',
+    actual: flyAppNameFor('abc123') === flyAppNameFor('abc123'),
```

Biome (2.5.6)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)